Core/HLE/sceNetAdhoc.cpp

```diff
@@ -47,8 +47,11 @@ static bool netAdhocMatchingInited;
 int netAdhocMatchingStarted = 0;
 
 SceUID threadAdhocID;
-Thread *threadAdhoc;
+
+std::vector<std::pair<u32, u32>> adhocctlEvents;
```

Comment:
These two queues (implemented as vectors but that's fine, although deque might be more appropriate) need to be protected by a mutex. It's fine for them to share one.

Simply add a global mutex here, and lock it around the emplace_backs and add a lock around the two "popping loops".